Utilities/Thread.cpp

```diff
@@ -765,7 +849,11 @@ static const decltype(REG_RAX) reg_table[] =
 };
 
 #define X64REG(context, reg) (&(context)->uc_mcontext.gregs[reg_table[reg]])
+#ifdef __sun
```

Comment:
> I feel like the bitrot here will get pretty bad.

When BSD would break so is likely Solaris. Fixing them together wouldn't significantly increase the support cost. It doesn't even need to be complete, NetBSD and OpenBSD are probably incomplete as well. I only maintain FreeBSD and DragonFly support. Keeping Solaris in mind helps to not forget `#else` case for strange or new platforms.

> BSD support I can get behind, but Solaris?

Third-party code already supports Solaris. PkgSrc provides recent enough toolchain.
As long as the OS can still be used on desktop in 2017 why not?

> The amount of BSD users using RPCS3 would be tiny anyway, but I think Solaris support is a lost cause.

I'm not adding Bitrig, that is most certainly dead. [In 2015](https://stats.wikimedia.org/wikimedia/squids/SquidReportOperatingSystems.htm) Solaris was 2.6x more popular than OpenBSD. YMMV